internal/pkg/scaffold/olm-catalog/csv_updaters.go

```diff
@@ -248,32 +249,45 @@ func (store *updaterStore) AddOwnedCRD(yamlDoc []byte) error {
 	if err := yaml.Unmarshal(yamlDoc, crd); err != nil {
 		return err
 	}
-	store.crds.Owned = append(store.crds.Owned, olmapiv1alpha1.CRDDescription{
+	crdDesc := olmapiv1alpha1.CRDDescription{
 		Name:    crd.ObjectMeta.Name,
 		Version: crd.Spec.Version,
 		Kind:    crd.Spec.Names.Kind,
-	})
-	store.crds.crKinds[crd.Spec.Names.Kind] = struct{}{}
+	}
+	store.crds.crKinds[crdDesc.Kind] = struct{}{}
+	// Parse CRD descriptors from source code comments and annotations.
+	gvk := schema.GroupVersionKind{Group: crd.Spec.Group, Version: crdDesc.Version, Kind: crdDesc.Kind}
+	if err := setCRDDescriptorForGVK(&crdDesc, gvk); err != nil {
+		return err
+	}
+	store.crds.Owned = append(store.crds.Owned, crdDesc)
 	return nil
 }
 
+// crdDescID produces an opaque, per-CSV unique string identifying
+// a CRDDescription.
+func crdDescID(desc olmapiv1alpha1.CRDDescription) string {
+	return desc.Version + desc.Kind
```

Comment:
While highly unlikely, there's no rule against having two owned CRDs in a CSV under different groups but the same version and Kind.

Which would make this not unique right? So any reason why we can't use the full gvk?
```Go
return desc.Name + desc.Version + desc.Kind
```
`Name` is usually set as the full group so far as I can tell.

Reply:
You're correct, it looks like [`desc.Name`](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md#required-crds) will always be the CRD's `metadata.name` which must be [`<lower kind>.<group>`](https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1beta1/types.go#L353-L354). We can actually remove the `name` element from its defining annotation [`customresourcedefinitions:resources="kind,version\"name\""`](https://github.com/operator-framework/operator-sdk/blob/e9e59c0/doc/user/csv-annotations.md#customresourcedefinitions) considering the name always has a defined format and we can retrieve it from a CRD.